libs/langchain/langchain/vectorstores/redis/redis.py

```diff
@@ -132,84 +99,113 @@ class Redis(VectorStore):
             )
     """
 
+    _default_content_vector_schema = {
+        "name": "content_vector",
+        "dims": 1536,
+        "distance_metric": "cosine",
+        "algorithm": "flat",
+        "datatype": "float32",
+    }
+
     def __init__(
         self,
         redis_url: str,
         index_name: str,
         embedding_function: Callable,
         content_key: str = "content",
-        metadata_key: str = "metadata",
         vector_key: str = "content_vector",
+        vector_schema: Optional[Dict] = None,
+        metadata_schema: Optional[Dict] = None,
         relevance_score_fn: Optional[Callable[[float], float]] = None,
-        distance_metric: REDIS_DISTANCE_METRICS = "COSINE",
         **kwargs: Any,
     ):
         """Initialize with necessary components."""
-        self.embedding_function = embedding_function
         self.index_name = index_name
+        self.content_key = content_key
+        self.vector_key = vector_key
+        self.embedding_function = embedding_function
         try:
             redis_client = get_client(redis_url=redis_url, **kwargs)
             # check if redis has redisearch module installed
-            _check_redis_module_exist(redis_client, REDIS_REQUIRED_MODULES)
+            check_redis_module_exist(redis_client, REDIS_REQUIRED_MODULES)
         except ValueError as e:
             raise ValueError(f"Redis failed to connect: {e}")
 
+        # Assume import check is done in get_client
+        from langchain.vectorstores.redis.schema import RedisMetadata
+
         self.client = redis_client
-        self.content_key = content_key
-        self.metadata_key = metadata_key
-        self.vector_key = vector_key
-        self.distance_metric = distance_metric
         self.relevance_score_fn = relevance_score_fn
+        self._vector_schema = self._default_content_vector_schema
+        if isinstance(vector_schema, dict):
+            self._vector_schema.update(vector_schema)
```

Comment:
i believe this will mutate the class attribute (_default_content_vector_schema)
![Screenshot 2023-08-02 at 6 51 14 PM](https://github.com/langchain-ai/langchain/assets/22008038/8e230d78-7874-4802-836f-dd8ec030367b)

Reply:
Great catch!